### PR TITLE
Added color variations on Text & Heading components

### PR DESCRIPTION
**Pull request by driesd**

### CODE REVIEW

File: components/typography/Heading.js [hunk 1]
```diff
@@ -8,17 +8,38 @@ const factory = (type, defaultElement) => {
...
       children: PropTypes.node,
       className: PropTypes.string,
+      color: PropTypes.oneOf(['white', 'neutral', 'mint', 'teal', 'violet', 'ruby', 'gold', 'aqua']),
       element: PropTypes.node,
+      soft: PropTypes.bool,
     };
 
     static defaultProps = {
       element: null,
+      soft: false,
```
Comment (nickwaelkens) [inline components/typography/Heading.js]: **PP** Prefer using `isCondition` signature for boolean values, eg. `isSoft`.

Comment (driesd) [reply components/typography/Heading.js]: As we recently discussed, we'd prefer to pass the props without the **is** prefix. But, functions that return a boolean value we should always prefix with **is**, **has**, ...

Comment (lowiebenoot) [reply components/typography/Heading.js]: +1, I prefer props without `is` as well, so you can do for example `<Heading soft>` (makes more sense than `<Heading isSoft>)`.

Comment (nickwaelkens) [reply components/typography/Heading.js]: Yeah, I was seriously confused, sorry boizzz 👍

File: components/typography/Heading.js [hunk 2]
```diff
@@ -8,17 +8,38 @@ const factory = (type, defaultElement) => {
     static propTypes = {
       children: PropTypes.node,
       className: PropTypes.string,
+      color: PropTypes.oneOf(['white', 'neutral', 'mint', 'teal', 'violet', 'ruby', 'gold', 'aqua']),
```
Comment (Sambego) [inline components/typography/Heading.js]: It looks like we're using this array of colors in a lot of places, maybe we can keep it somewhere separate and reuse it in all components. Should there ever be the need to add a new color, we have to add it in just one place.

**Comment by Sambego** [comment]
The `Heading` and `Text` factories are completely the same, except the word "heading" and "text". Why no use a single factory to create these, instead of duplicating code?

screen shot 2017-10-12 at 10 44 56

**Comment by svenlombaert** [comment]
I'm completely on the same track as @Sambego with this. Header is indeed a variation on the text component.

File: components/typography/Heading.js (hunk 2 above)
Comment (driesd) [reply components/typography/Heading.js]: Yes, we could do that, but not always the same color variations are applicable to our components.

**Comment by driesd** [comment]
You are both right, but they get a different classname (.header vs .text) and if you take a look in ui-typography you'll notice that .heading & .text got slightly different styling.

**Pull request merged**